desktop/main.js

```diff
@@ -36,25 +36,40 @@ function pasteAsPlainText(browserWindow) {
     browserWindow.webContents.insertText(text);
 }
 
-// Initialize the right click menu
-// See https://github.com/sindresorhus/electron-context-menu
-// Add the Paste and Match Style command to the context menu
-contextMenu({
-    append: (defaultActions, parameters, browserWindow) => [
-        new MenuItem({
-            // Only enable the menu item for Editable context which supports paste
-            visible: parameters.isEditable && parameters.editFlags.canPaste,
-            role: 'pasteAndMatchStyle',
-            accelerator: DESKTOP_SHORTCUT_ACCELERATOR.PASTE_AND_MATCH_STYLE,
-        }),
-        new MenuItem({
-            label: Localize.translate(CONST.LOCALES.DEFAULT, 'desktopApplicationMenu.pasteAsPlainText'),
-            visible: parameters.isEditable && parameters.editFlags.canPaste && clipboard.readText().length > 0,
-            accelerator: DESKTOP_SHORTCUT_ACCELERATOR.PASTE_AS_PLAIN_TEXT,
-            click: () => pasteAsPlainText(browserWindow),
-        }),
-    ],
-});
+/**
+ * Initialize the right-click menu
+ * See https://github.com/sindresorhus/electron-context-menu
+ *
+ * @param {string} preferredLocale - The current user language to be used for translating menu labels.
+ * @returns {Function} A dispose function to clean up the created context menu.
+ */
+
+function createContextMenu(preferredLocale) {
+    return contextMenu({
+        labels: {
+            cut: Localize.translate(preferredLocale, 'desktopApplicationMenu.cut'),
+            paste: Localize.translate(preferredLocale, 'desktopApplicationMenu.paste'),
+            copy: Localize.translate(preferredLocale, 'desktopApplicationMenu.copy'),
+        },
+        append: (defaultActions, parameters, browserWindow) => [
+            new MenuItem({
+                // Only enable the menu item for Editable context which supports paste
+                visible: parameters.isEditable && parameters.editFlags.canPaste,
+                role: 'pasteAndMatchStyle',
+                accelerator: DESKTOP_SHORTCUT_ACCELERATOR.PASTE_AND_MATCH_STYLE,
+                label: Localize.translate(preferredLocale, 'desktopApplicationMenu.pasteAndMatchStyle'),
+            }),
+            new MenuItem({
+                label: Localize.translate(preferredLocale, 'desktopApplicationMenu.pasteAsPlainText'),
+                visible: parameters.isEditable && parameters.editFlags.canPaste && clipboard.readText().length > 0,
+                accelerator: DESKTOP_SHORTCUT_ACCELERATOR.PASTE_AS_PLAIN_TEXT,
+                click: () => pasteAsPlainText(browserWindow),
+            }),
+        ],
+    });
+}
+
+let disposeContextMenu = createContextMenu(CONST.LOCALES.DEFAULT);
```

Comment:
Initialization is already done here, isn't it?

Reply:
Yes, this will initialize the menu. And the below will recreate it when the user changes the language 
![image](https://github.com/Expensify/App/assets/65986357/83797832-7760-47b6-8400-e87bbfce2c43)

Reply:
Still trying to understand.

createContextMenu is called 3 times total right?

Reply:
2 times

Reply:
- `let disposeContextMenu = createContextMenu(CONST.LOCALES.DEFAULT);` calls `createContextMenu` app start

after changing language:
- `disposeContextMenu();` calls `createContextMenu`
- and then `disposeContextMenu = createContextMenu(updatedLocale);` calls `createContextMenu` again

Reply:
`disposeContextMenu` will not call `createContextMenu`. It will call the returned value of `createContextMenu` which will dispose the event listeners 
> The return value of contextMenu() is a function that disposes of the created event listeners
https://www.npmjs.com/package/electron-context-menu

Reply:
`contextMenu`  function inside `createContextMenu` creates the actual contextMenu

Reply:
The `disposeContextMenu` is a function like the one you return in `useEffect` to clear the event listeners

Reply:
Got it. Clear now. Thanks